### PR TITLE
feat: Disable scrape metrics by default

### PR DESCRIPTION
**Pull request by hisarbalik**
## Description

Changes proposed in this pull request (what was done and why):

- By default scrape metrics disabled for MetricPipeline inputs `istio` and `prometheus`
- Add new configuration section to the MetricPipeline inputs `istio` and `prometheus` to manage scrape metrics (enable or disable) per input.
- Update user documents about new MetricPipeline configuration sections

Changes refer to particular issues, PRs or documents:

- https://github.com/kyma-project/telemetry-manager/issues/661

## Traceability
- [x] The PR is linked to a GitHub issue.
- [ ] New features have a milestone set.
- [x] New features have defined acceptance criteria in a corresponding GitHub Issue, and all criteria are satisfied with this PR.
- [x] The corresponding GitHub issue has a respective `area` and `kind` label.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.
- [x] Adjusted the documentation if the change is user-facing.
- [x] The feature is unit-tested
- [x] The feature is e2e-tested

### CODE REVIEW

File: apis/telemetry/v1alpha1/metricpipeline_types.go [hunk 1]
```diff
@@ -87,6 +87,8 @@ type MetricPipelinePrometheusInput struct {
 	//+optional
 	//+kubebuilder:default={exclude: {kyma-system, kube-system, istio-system, compass-system}}
 	Namespaces *MetricPipelineInputNamespaceSelector `json:"namespaces,omitempty"`
+	// Configures diagnostic metrics scraping
```
Comment (rakesh-garimella) [inline apis/telemetry/v1alpha1/metricpipeline_types.go:90]: should we make it optional as well ?

File: docs/user/04-metrics.md [hunk 2]
```diff
@@ -418,8 +418,60 @@ spec:
...
 
-### Step 9: Deploy the Pipeline
+When using the `prometheus` or `istio` input feature of the MetricPipeline, for every metric source typical scrape metrics are getting produced and these are concrete
+- up
+- scrape_duration_seconds
+- scrape_samples_scraped
+- scrape_samples_post_metric_relabeling
+- scrape_series_added
+
+These are rather technical metrics and useful for debugging and diagnose purposes. However, they are also causing data and they are by default disabled.
```
Comment (rakesh-garimella) [inline docs/user/04-metrics.md:430]:
```suggestion
These are rather technical metrics and useful for debugging and diagnostic purposes. 
```

may be we can write this whole sentence in a better way highlighting the additional space requirement.

File: internal/otelcollector/config/metric/gateway/processors.go [hunk 3]
```diff
@@ -79,6 +79,19 @@ func makeDropIfInputSourceOtlpConfig() *FilterProcessor {
 	}
 }
 
+func makeDropDiagnosticMetricsForInput(inputSourceCondition string) *FilterProcessor {
```
Comment (rakesh-garimella) [inline internal/otelcollector/config/metric/gateway/processors.go:82]: can we move it to a separate file like this: https://github.com/kyma-project/telemetry-manager/blob/main/internal/otelcollector/config/metric/agent/drop_internal_communication.go and have a test ?

File: test/integration/istio/metrics_istio_input_test.go [hunk 4]
```diff
@@ -215,6 +215,17 @@ var _ = Describe("Metrics Istio Input", Label("metrics"), func() {
 		It("Should verify that istio metric with destination_workload=telemetry-metric-gateway does not exist", func() {
 			verifyMetricIsNotPresent(telemetryExportURL, "destination_workload", "telemetry-metric-gateway")
 		})
+
+		It("Ensures no diagnostic metrics are sent to backend", func() {
+			Eventually(func(g Gomega) {
```
Comment (rakesh-garimella) [inline test/integration/istio/metrics_istio_input_test.go:220]: use `consistently` instead of `eventually`

**Review by rakesh-garimella — COMMENTED** [review]
Please review the comments

File: config/samples/telemetry_v1alpha1_metricpipeline.yaml [hunk 5]
```diff
@@ -161,4 +161,17 @@ spec:
   output:
     otlp:
       endpoint:
-        value: http://mock-metric-receiver.metric-receiver:4317
+        value: ingest-otlp-sf-1502277b-af0e-44ef-b169-a4d004105616.cls-01.cloud.logs.services.sap.hana.ondemand.com:443
```
Comment (rakesh-garimella) [inline config/samples/telemetry_v1alpha1_metricpipeline.yaml:164]: please revert

File: apis/telemetry/v1alpha1/metricpipeline_types.go (hunk 1 above)
Comment (hisarbalik) [reply apis/telemetry/v1alpha1/metricpipeline_types.go:90]: optional annotation added

File: internal/otelcollector/config/metric/gateway/processors.go (hunk 3 above)
Comment (hisarbalik) [reply internal/otelcollector/config/metric/gateway/processors.go]: separate unit test added

**Review by chrkl — APPROVED** [review]

File: docs/user/04-metrics.md [hunk 6]
```diff
@@ -418,8 +418,60 @@ spec:
 
 Note that metrics from system namespaces are excluded by default when a namespace selector for the `prometheus` or `runtime` input is not defined. However, for the `istio` and `otlp` input, metrics from system namespaces are included by default if the namespace selector is not defined.
 
+### Step 9: Enable Diagnostic Metrics
 
-### Step 9: Deploy the Pipeline
+When using the `prometheus` or `istio` input feature of the MetricPipeline, for every metric source typical scrape metrics are getting produced and these are concrete
```
Comment (nataliasitko) [inline docs/user/04-metrics.md:423]:
```suggestion
When using the `prometheus` or `istio` input feature of the MetricPipeline, typical scrape metrics are produced for every metric source. These metrics include:
```

File: docs/user/04-metrics.md [hunk 7]
```diff
@@ -418,8 +418,60 @@ spec:
...
+- up
+- scrape_duration_seconds
+- scrape_samples_scraped
+- scrape_samples_post_metric_relabeling
+- scrape_series_added
+
+These are rather technical metrics and useful for debugging and diagnostic purposes.
```
Comment (nataliasitko) [inline docs/user/04-metrics.md:430]:
```suggestion
These are rather technical metrics, useful for debugging and diagnostic purposes. 
```

File: docs/user/04-metrics.md [hunk 8]
```diff
@@ -418,8 +418,60 @@ spec:
...
 
+### Step 9: Enable Diagnostic Metrics
 
-### Step 9: Deploy the Pipeline
+When using the `prometheus` or `istio` input feature of the MetricPipeline, for every metric source typical scrape metrics are getting produced and these are concrete
+- up
+- scrape_duration_seconds
+- scrape_samples_scraped
+- scrape_samples_post_metric_relabeling
+- scrape_series_added
```
Comment (nataliasitko) [inline docs/user/04-metrics.md:428]:
```suggestion
- `up`
- `scrape_duration_seconds`
- `scrape_samples_scraped`
- `scrape_samples_post_metric_relabeling`
- `scrape_series_added`
```

File: docs/user/04-metrics.md [hunk 9]
```diff
@@ -418,8 +418,60 @@ spec:
...
+      enabled: true
+      diagnosticMetrics:
+        enabled: true
+  output:
+    otlp:
+      endpoint:
+        value: https://backend.example.com:4317
+```
+
+Note diagnostic metrics are available only for inputs `prometheus` adn `istio` and are disabled by default.
```
Comment (nataliasitko) [inline docs/user/04-metrics.md:472]:
```suggestion
Diagnostic metrics are only available for inputs `prometheus` and `istio`. They are disabled by default.
```

**Review by chrkl — APPROVED** [review]

**Pull request merged**